lib/src/main/java/com/otaliastudios/transcoder/internal/thumbnails/DefaultThumbnailsEngine.kt

```diff
@@ -250,10 +250,31 @@ class DefaultThumbnailsEngine(
             positions.mapNotNull { (positionUs, request) ->
                 val localizedUs = timer.localize(TrackType.VIDEO, index, positionUs)
                 localizedUs?.let { Stub(request, positionUs, localizedUs) }
-            }.toMutableList().sortedBy { it.positionUs }
+//            }.toMutableList().sortedBy { it.positionUs }
+            }.toMutableList().reorder(dataSources[TrackType.VIDEO][0])
         )
     }
 
+    private fun  List<Stub>.reorder(source: DataSource): Collection<Stub> {
+        val bucketListMap = LinkedHashMap<Long, ArrayList<Stub>>()
+        val finalList = ArrayList<Stub>()
+
+        forEach {
+            val nextKeyFrameIndex = source.search(it.positionUs)
+            val previousKeyFrameUs =
+                source.keyFrameTimestampsUsList[
```

Comment:
Can be replaced with another extension method.
like source.getKeyFrameAt(pos: Int)

Reply:
Basically, the list shouldn't be exposed outside.